src/qt/rpcconsole.cpp

```diff
@@ -464,6 +454,96 @@ void RPCExecutor::request(const QString &command, const WalletModel* wallet_mode
+    // parsed_command should be the same size or longer than parsed_key because it might have parameters
+    return parsed_key.size() <= parsed_command.size() && std::equal(parsed_key.begin(), parsed_key.end(), parsed_command.begin());
+}
+
+/**
+ * @brief  Executes the console-only command "help-console".
+ * @param  parsed_command  A vector of strings with command and parameters, usually generated by RPCExecutor::parseHelper
+ * @param wallet_model  WalletModel to use for the command
+ * @return  True if the command was executed, false otherwise.
+ */
+bool RPCExecutor::executeConsoleHelpConsole(const std::vector<std::string>& parsed_command, const WalletModel* wallet_model)
+{
+    // Catch the console-only-help command before RPC call is executed and reply with help text as-if a RPC reply.
+    if (!parsed_command.empty() && parsed_command[0] == "help-console") {
+        Q_EMIT reply(RPCConsole::CMD_REPLY,
+                     QString("\n"
+                             "This console accepts RPC commands using the standard syntax.\n"
+                             "   example:    getblockhash 0\n\n"
+                             "This console can also accept RPC commands using the parenthesized syntax.\n"
+                             "   example:    getblockhash(0)\n\n"
+                             "Commands may be nested when specified with the parenthesized syntax.\n"
+                             "   example:    getblock(getblockhash(0) 1)\n\n"
+                             "A space or a comma can be used to delimit arguments for either syntax.\n"
+                             "   example:    getblockhash 0\n"
+                             "               getblockhash,0\n\n"
+                             "Named results can be queried with a non-quoted key string in brackets using the parenthesized syntax.\n"
+                             "   example:    getblock(getblockhash(0) 1)[tx]\n\n"
+                             "Results without keys can be queried with an integer in brackets using the parenthesized syntax.\n"
+                             "   example:    getblock(getblockhash(0),1)[tx][0]\n\n"));
+        return true;
+    } else {
+        return false;
+    }
+}
+
+/**
+ * Catches console-only command before a RPC call is executed
+ *
+ * @param[in]    command  Command line to execute
+ * @param[in]    wallet_model  Wallet model to use
+ * @return  true if command was handled by this method (even on errors), false otherwise
+ *
+ */
+bool RPCExecutor::executeConsoleOnlyCommand(const std::string& command, const WalletModel* wallet_model)
+{
+    // Parse command line into a vector of strings
+    const std::vector<std::string> parsed_command{parseHelper(command)};
+    if (parsed_command.empty()) {
+        return false;
+    } else {
+        // Iterate over m_method_map and execute the associated method if the beggining of parsed_command matches the key
```

Comment:
In ad5642ae:

I think that would be cleaner to change the if/else with a:

```c++
const std::vector<std::string> parsed_command{parseHelper(command)};
if (parsed_command.empty()) return false;

// Iterate over m_method_map and execute the associated method if the beggining of parsed_command matches the key
etc..
```

Reply:
Agree ! Changed